patches/spec.fetched.json/08-make-vrf-required-param-in-vrf-metal-gateway.patch

```diff
@@ -1,10 +1,69 @@
+diff --git a/spec/oas3.patched/openapi/public/components/schemas/IPReservationOrHref.yaml b/spec/oas3.patched/openapi/public/components/schemas/IPReservationOrHref.yaml
+new file mode 100644
+index 0000000..42bb973
+--- /dev/null
++++ b/spec/oas3.patched/openapi/public/components/schemas/IPReservationOrHref.yaml
+@@ -0,0 +1,3 @@
++anyOf:
++- $ref: './IPReservation.yaml'
++- $ref: './Href.yaml'
+diff --git a/spec/oas3.patched/openapi/public/components/schemas/MetalGateway.yaml b/spec/oas3.patched/openapi/public/components/schemas/MetalGateway.yaml
+index ed0eb30..64aa9a5 100644
+--- a/spec/oas3.patched/openapi/public/components/schemas/MetalGateway.yaml
++++ b/spec/oas3.patched/openapi/public/components/schemas/MetalGateway.yaml
+@@ -10,7 +10,7 @@ properties:
+     format: uuid
+     type: string
+   ip_reservation:
+-    $ref: './IPReservation.yaml'
++    $ref: './IPReservationOrHref.yaml'
+   project:
+     $ref: './Project.yaml'
+   state:
+@@ -30,3 +30,4 @@ properties:
+   virtual_network:
+     $ref: './VirtualNetwork.yaml'
+ type: object
++additionalProperties: false
```

Comment:
This is OK for unblocking, although we may go in a different direction for the upstream, since this takes some flexibility away from the MetalGateway model.

Reply:
One alternative workaround could be to add an invalid definition for the `vrf` property in the `MetalGateway` model, something like:

```yaml
vrf:
  description: This attribute will always be null for a Metal Gateway that is not in a VRF
  type: string
```

Since the `vrf` property will not be a string, that should cause the VRF Metal Gateway JSON to fail validation against the non-VRF Metal Gateway model.